src/main/java/org/commcare/session/CommCareSession.java

```diff
@@ -116,74 +117,92 @@ public CommCareSession(CommCareSession oldCommCareSession) {
         }
     }
 
-    private Vector<Entry> getEntriesForCommand(String commandId) {
-        return getEntriesForCommand(commandId, new OrderedHashtable<String, String>());
+    public Vector<Entry> getEntriesForCommand(String commandId) {
+        return getEntriesForCommand(commandId, false);
+    }
+
+    public Vector<Entry> getEntriesForCommand(String commandId, boolean includeNested) {
+        return getEntriesForCommand(commandId, new OrderedHashtable<String, String>(), includeNested);
     }
 
     /**
      * @param commandId          the current command id
      * @param currentSessionData all of the datums already on the stack
+     * @param includeNested     whether to include entries from child modules
      * @return A list of all of the form entry actions that are possible with the given commandId
      * and the given list of already-collected datums
      */
     private Vector<Entry> getEntriesForCommand(String commandId,
-                                               OrderedHashtable<String, String> currentSessionData) {
+                                               OrderedHashtable<String, String> currentSessionData,
+                                               boolean includeNested) {
         Vector<Entry> entries = new Vector<>();
         if (commandId == null) {
             return entries;
         }
         for (Suite s : platform.getInstalledSuites()) {
             List<Menu> menusWithId = s.getMenusWithId(commandId);
             if (menusWithId != null) {
-                for (Menu menu : menusWithId) {
-                    entries.addAll(getEntriesFromMenu(menu, currentSessionData));
+                // make a copy so that we can use this as a queue
+                List<Menu> menusToExamine = new ArrayList<>(menusWithId);
+                while (!menusToExamine.isEmpty()) {
+                    Menu menu = menusToExamine.remove(0);
```

Comment:
Super minor point that if this was a Stack you could pop(), or a queue you could poll() and it would be a bit more clear how you're using the datatype.

Reply:
good call